rerun_py/rerun_sdk/rerun/components/line_strip2d_ext.py

```diff
@@ -46,18 +46,17 @@ def native_to_pa_array_override(data: LineStrip2DArrayLike, data_type: pa.DataTy
             else:
                 # Is it a single strip or several?
                 # It could be a sequence of the style `[[0, 0], [1, 1]]` which is a single strip.
-                if all(
-                    isinstance(elem, Sequence) and len(elem) > 0 and isinstance(elem[0], numbers.Number)
-                    for elem in data
-                ):
-                    if all(len(elem) == 2 for elem in data):  # type: ignore[arg-type]
+                if isinstance(data[0], Sequence) and len(data[0]) > 0 and isinstance(data[0][0], numbers.Number):
+                    if len(data[0]) == 2:  # type: ignore[arg-type]
+                        # If any of the following elements are not sequence of length 2, Vec2DBatch should raise an error.
```

Comment:
Vec2DBatch will swallow the error thanks to our exception handling policy, but: https://github.com/rerun-io/rerun/pull/3672 will fix this for you.